src/widget.js

```diff
@@ -1,23 +1,140 @@
-var buttonUrl = "https://superhero.com/buttons/v1/";
-document.addEventListener("DOMContentLoaded", function(event) {
-  var buttons = document.querySelectorAll("a.superhero-tip-button");
-  var iframes = [];
-  for (var i = 0; i < buttons.length; i++) {
-    var size = buttons[i].getAttribute("data-size") ? buttons[i].getAttribute("data-size") : "icon";
-    var href = buttons[i].getAttribute("data-href") ? buttons[i].getAttribute("data-href") : window.location.href;
-    var account = buttons[i].getAttribute("data-account");
-
-    var iframe = document.createElement('iframe');
-    iframe.src = buttonUrl + size + "?url=" + href;
-    iframe.setAttribute('style','max-width:210px;max-height:120px;overflow:hidden;');
-    iframe.setAttribute("scrolling", "no");
-    iframe.setAttribute("frameborder", "0");
-    iframe.className = "superhero-tip-button";
-    if (account) iframe.setAttribute("data-account", account);
-    iframes[i] = iframe;
+const { detect } = require('detect-browser');
+
+const defaults = {
+  size: 'icon',
+  url: window.location.href,
+  inlineCss: false,
+  version: 1,
+  all: false
+};
+
+const templates = {
+  icon: `<div class="superhero-button-link-holder">
+            <a id="superhero-tip-link" class="superhero-button-link" target="_blank">
+              <img src="../img/v1-icon.png" />
+            </a>
+          </div>`,
+  small: `<div class="superhero-button-link-holder">
+            <a id="superhero-tip-link" class="superhero-button-link" target="_blank">
+              <img src="../img/v1-icon.png" />
+              <span>Donate</span>
+            </a>
+          </div>
+          <div class="superhero-button-tips-amount"><span id="tips">0</span><span class="ae">AE</span></div>`,
+  medium: `<div class="superhero-button-link-holder">
+            <a id="superhero-tip-link" class="superhero-button-link" target="_blank">
+              <img src="../img/v1-icon.png" />
+              <span>Donate Now</span>
+            </a>
+          </div>
+          <div class="superhero-button-tips-amount"><span id="tips">0</span><span class="ae">AE</span></div>`,
+  large: `<div class="superhero-button-tips-amount"><span id="tips">0</span><span class="ae">AE</span></div>
+          <div class="superhero-button-link-holder">
+            <a id="superhero-tip-link" class="superhero-button-link" target="_blank">
+              <img src="../img/v1-icon.png" />
+              <span>Donate Now</span>
+            </a>
+          </div>`
+};
+
+class SuperheroButton {
+  
+  constructor(element, options) {
+    const opt = { ...defaults, ...options };
+    this.insertStyles(opt);
+    const el = this.isDomNode(element) || this.isNodeList(element) ? element : opt.all ? document.querySelectorAll(element) : document.querySelector(element);
```

Comment:
I propose to drop `all` option (developer may use a more precise selector instead or at least call querySelector by himself) and the reverse check would look clearer:
```suggestion
    const el = typeof element === 'string' ? document.querySelectorAll(element) : element;
```